### PR TITLE
Merchant deletes an item userstory47 #25

### LINKED ISSUE #25: User Story 47, Merchant deletes an item
```
As a merchant
When I visit my items page
And I click on a "delete" button or link for an item
I am returned to my items page
I see a flash message indicating this item is now deleted
I no longer see this item on the page
```

DO THESE STORIES FIRST:
- User Story 44, Merchant Items Index Page

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #29

child of #82

depends on #28

### PR DESCRIPTION
**Pull request by milevy1**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100%
- rspec spec/models - 98.28%

# What functionality does this accomplish?
User Story 47, Merchant deletes an item
- Merchants can now delete items (as long as they have no orders)

# What did you struggle on to complete?

# Implements/Fixes:
* description
closes #25 

# Testing Changes
- [x] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

# Please include an emoji of how you feel about this branch:
# 💯

### CODE REVIEW

File: spec/features/merchants/items_index_spec.rb [hunk 1]
```diff
@@ -99,5 +99,17 @@
       @inactive_item.reload   # .reload in order to update the object with new database value
       expect(@inactive_item.enabled).to eq(true)
     end
+
+    scenario 'Clicking on Delete link for an item disables' do
```
Comment (blake-enyart) [inline spec/features/merchants/items_index_spec.rb]: nice use of scenario over context here.
